addons/l10n_dk_oioubl/models/account_edi_xml_oioubl_201.py

```diff
@@ -93,9 +152,12 @@ def _export_invoice_vals(self, invoice):
     def _get_partner_party_vals(self, partner, role):
         # EXTENDS account.edi.xml.ubl_20
         vals = super()._get_partner_party_vals(partner, role)
-        if partner.peppol_endpoint:
+        if partner.peppol_endpoint and partner.peppol_eas in PEPPOL_ENDPOINT_MAPPING:
+            # if we don't know the mapping with real names for the Peppol Endpoint, fallback on VAT simply
             endpoint_id = partner.peppol_endpoint
-            scheme_id = 'GLN'
+            scheme_id = PEPPOL_ENDPOINT_MAPPING[partner.peppol_eas]
+            if (scheme_id in ('DK:CVR', 'FR:SIRET') or scheme_id[3:] == 'VAT') and endpoint_id[:2].isnumeric():
```

Comment:
```suggestion
            if (scheme_id in ('DK:CVR', 'FR:SIRET') or scheme_id[3:] == 'VAT') and endpoint_id.isnumeric():
```
Could be done on the full string, it's pretty confusing here :sweat_smile: